code/modules/projectiles/projectile/special.dm

```diff
@@ -257,3 +257,84 @@
 			visible_message("<span class='danger'>\The [src] splatters a layer of web on \the [target]!</span>")
 			new /obj/effect/spider/stickyweb(target.loc)
 	..()
+
+/obj/item/projectile/beam/tungsten
+	name = "core of molten tungsten"
+	icon_state = "energy"
+	fire_sound = 'sound/weapons/emitter2.ogg'
+	pass_flags = PASSTABLE | PASSGRILLE
+	damage = 70
+	damage_type = BURN
+	check_armour = "laser"
+	light_range = 4
+	light_power = 3
+	light_color = "#3300ff"
+
+	muzzle_type = /obj/effect/projectile/tungsten/muzzle
+	tracer_type = /obj/effect/projectile/tungsten/tracer
+	impact_type = /obj/effect/projectile/tungsten/impact
+
+/obj/item/projectile/beam/tungsten/on_hit(var/atom/target, var/blocked = 0)
+	if(isliving(target))
+		var/mob/living/L = target
+		L.add_modifier(/datum/modifier/grievous_wounds, 30 SECONDS)
+		if(ishuman(L))
+			var/mob/living/carbon/human/H = L
+
+			var/target_armor = H.getarmor(def_zone, check_armour)
+			var/obj/item/organ/external/target_limb = H.get_organ(def_zone)
+
+			var/armor_special = 0
+
+			if(target_armor >= 60)
+				var/list/throw_dirs = list(1, 2, 4, 8, 5, 6, 9, 10)

```

Comment:
`var/turf/T = get_step(H, pick(alldirs - src.dir))`
`alldirs` is a global list containing the human-readable dirs:
`var/list/alldirs = list(NORTH, SOUTH, EAST, WEST, NORTHEAST, NORTHWEST, SOUTHEAST, SOUTHWEST)`